addons/sale_timesheet/report/timesheets_analysis_views.xml

```diff
@@ -50,6 +50,41 @@
         </field>
     </record>
 
+    <record id="hr_timesheet_report_search_sale_timesheet" model="ir.ui.view">
+        <field name="name">timesheets.analysis.report.search</field>
+        <field name="model">timesheets.analysis.report</field>
+        <field name="inherit_id" ref="hr_timesheet.hr_timesheet_report_search"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='task_id']" position="after">
+                <field name="order_id" string="Sales Order" groups="sales_team.group_sale_salesman"/>
+                <field name="so_line" groups="sales_team.group_sale_salesman"/>
+            </xpath>
+            <xpath expr="//filter[@name='month']" position="before">
+                <filter name="billable_fixed" string="Billed at a Fixed Price" domain="[('timesheet_invoice_type', '=', 'billable_fixed')]"
+                    groups="sales_team.group_sale_salesman"/>
+                <filter name="billable_time" string="Billed on Timesheets" domain="[('timesheet_invoice_type', '=', 'billable_time')]"
+                    groups="sales_team.group_sale_salesman"/>
+                <filter name="billable_milestones" string="Billed on Milestones" domain="[('timesheet_invoice_type', '=', 'billable_milestones')]"
+                    groups="sales_team.group_sale_salesman"/>
+                <filter name="billable_manual" string="Billed Manually" domain="[('timesheet_invoice_type', '=', 'billable_manual')]"
+                    groups="sales_team.group_sale_salesman"/>
+                <filter name="non_billable" string="Non-Billable" domain="[('timesheet_invoice_type', '=', 'non_billable')]"
+                    groups="sales_team.group_sale_salesman"/>
+                <separator/>
+            </xpath>
+            <xpath expr="//filter[@name='groupby_employee']" position="after">
+                <filter string="Sales Order" name="groupby_sale_order" domain="[]"
+                    context="{'group_by': 'order_id'}" groups="sales_team.group_sale_salesman"/>
+                <filter string="Sales Order Item" name="groupby_sale_order_item" domain="[]"
+                    context="{'group_by': 'so_line'}" groups="sales_team.group_sale_salesman"/>
+                <filter string="Invoice" name="groupby_invoice" domain="[]"
+                    context="{'group_by': 'timesheet_invoice_id'}" groups="sales_team.group_sale_salesman"/>
+                <filter string="Billable Type" name="groupby_timesheet_invoice_type" domain="[]"
+                    context="{'group_by': 'timesheet_invoice_type'}" groups="sales_team.group_sale_salesman"/>
+            </xpath>
+        </field>
```

Comment:
Just this?
```suggestion
        <field name="inherit_id" ref="sale_timesheet.timesheet_view_search"/>
        <field name="mode">primary</field>
        <field name="arch" type="xml">
            <!-- I think you need something in here for the view to be valid, so just a no-op action, to check if needed -->
            <xpath expr="//field[@name='task_id']" position="after"/>
        </field>
```

Reply:
Looks like at least the order of loading needs to be different :sweat_smile: 
(I didn't test the suggestion)